src/wsdiscovery.ts

```diff
@@ -27,92 +36,176 @@ export default class WsDiscovery extends EventEmitter {
-                if (this.ifStateConnected) {
-                    // Close socket only if it was bound and the interface was deactivated
-                    Logger.debug(
-                        `Network interface [${interfaceName}] removed. Bring interface up to rediscover Huddly network cameras`,
-                        WsDiscovery.name
-                    );
-                    this.socket.close();
-                }
-                this.ifStateConnected = false;
-            }
-        }, 1000);
+
+        if (networkInterfaces.linkLocalMaps.length > 0) {
+            networkInterfaces.linkLocalMaps.forEach(map => {
+                this.socketConnections[map.interfaceName] = this.initSocketConnection(map);
+            });
+        } else {
+            Logger.debug(`Unable to find any link local networks.`, WsDiscovery.name);
+        }
     }
 
-    findL1HostInterface(): any {
-        const interfaceMap = networkInterfaces();
-        const map = { ip: undefined, interface: undefined };
-        for (const [k, v] of Object.entries(interfaceMap)) {
+    findNetworkInterfaces(): DiscoveredInterfaces {
+        const interfaces = networkInterfaces();
+        const targetInterface: InterfaceMap = { interfaceName: undefined, ip: undefined };
+        const linkLocalMaps = [];
+        for (const [k, v] of Object.entries(interfaces)) {
             if (v instanceof Array) {
                 v.forEach((networkInterface: any) => {
                     // Make sure we operate on Ipv4 addresses only
                     if (networkInterface.family === 'IPv4') {
                         // Check if the integrator has specified the target interface or one of the interfaces is huddly compatible (BASE)
-                        if (
-                            (this.opts.targetInterfaceName && this.opts.targetInterfaceName == k) ||
-                            this.manufacturerFromMac(networkInterface.mac)
-                        ) {
-                            Logger.debug(
-                                `Discovery probe messages bound to interface ${k} on addr ${networkInterface.address}`,
-                                WsDiscovery.name
-                            );
-                            map.ip = networkInterface.address;
-                            map.interface = k;
+                        if (this.opts.targetInterfaceName || this.opts.targetInterfaceAddr) {
+                            if (
+                                this.opts.targetInterfaceName == k ||
+                                this.opts.targetInterfaceAddr == networkInterface.address
+                            ) {
+                                targetInterface.ip = networkInterface.address;
+                                targetInterface.interfaceName = k;
+                            }
+                        } else if (this.isLinkLocalAddr(networkInterface.address)) {
```

Comment:
I guess this condition here applies to BASE and when we have l1 connected through a PoE. But it wont exclude other link-local interfaces which i guess is ok. Just thinking out loud :)

Reply:
Yes, just for simplicity sake I've done it like this now. Also makes it behave as expected for people testing with PoE.